test/functional/ui/screen_multigrid_spec.lua

```diff
@@ -0,0 +1,704 @@
+    ## grid 2
+      ^                                                     |
+      {1:~                                                    }|
+      {1:~                                                    }|
+      {1:~                                                    }|
+      {1:~                                                    }|
+      {1:~                                                    }|
+      {1:~                                                    }|
+      {1:~                                                    }|
+      {1:~                                                    }|
+      {1:~                                                    }|
+      {1:~                                                    }|
+      {1:~                                                    }|
+    ]])
+  end)
+
+  it('positions windows correctly', function()
+    command('vsplit')
+    screen:expect([[
+    ## grid 1
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+      {11:[No Name]                  }{12:[No Name]                 }|
+                                                           |
+    ## grid 2
+                                |
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+    ## grid 3
+      ^                          |
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+    ]], nil, nil, function()
+      iswin(screen.win_position[2].win)
+      eq(0, screen.win_position[2].startrow)
```

Comment:
these could be written more compactly (and consistent with existing tests) as
```
eq({
  [2]={win={id=1001},startrow=0,...},
  [3]={win=...},
}, screen.win_position)
```
as a bonus, this will automatically assert that no spurious `win_position` was sent with unexpected grid ids (and make sure we cover every case if we decide to add a new key).